packages/mui-material/src/Dialog/Dialog.js

```diff
@@ -282,56 +285,101 @@ const Dialog = React.forwardRef(function Dialog(inProps, ref) {
     return { titleId: ariaLabelledby };
   }, [ariaLabelledby]);
 
+  const backwardCompatibleSlots = {
+    transition: TransitionComponent,
+    ...slots,
+  };
+
+  const backwardCompatibleSlotProps = {
+    transition: TransitionProps,
+    paper: PaperProps,
+    backdrop: BackdropProps,
+    ...slotProps,
+  };
+
+  const externalForwardedProps = {
+    slots: backwardCompatibleSlots,
+    slotProps: backwardCompatibleSlotProps,
+  };
+
+  const [RootSlot, rootSlotProps] = useSlot('root', {
+    elementType: DialogRoot,
+    externalForwardedProps,
+    ownerState,
+    className: clsx(classes.root, className),
+    ref,
+  });
+
+  const [BackdropSlot, { as: backdropComponent, ...backdropSlotProps }] = useSlot('backdrop', {
+    elementType: DialogBackdrop,
+    externalForwardedProps,
+    ownerState,
+  });
```

Comment:
```suggestion
  const [BackdropSlot, backdropSlotProps] = useSlot('backdrop', {
    shouldForwardComponentProp: true,
    elementType: DialogBackdrop,
    externalForwardedProps,
    ownerState,
  });
```
With `shouldForwardComponentProp: true`, the `backdropSlotProps` will not produce `as` prop.